console-subscriber/src/aggregator.rs

```diff
@@ -131,26 +164,14 @@ impl Aggregator {
                 // a new client has started watching!
                 subscription = self.rpcs.recv() => {
                     if let Some(subscription) = subscription {
-                        tracing::debug!("new subscription");
-                        let new_tasks = self.tasks.all().map(|(id, task)| {
-                            task.to_proto(id.clone())
-                        }).collect();
-                        let now = SystemTime::now();
-                        let stats_update = self.stats.all().map(|(id, stats)| {
-                            (id.into_u64(), stats.to_proto())
-                        }).collect();
-                        // Send the initial state --- if this fails, the subscription is
-                        // already dead.
-                        if subscription.update(&proto::tasks::TaskUpdate {
-                            new_metadata: Some(proto::RegisterMetadata {
-                                metadata: self.all_metadata.clone(),
-                            }),
-                            new_tasks,
-                            stats_update,
-                            now: Some(now.into()),
-                        }) {
-                            self.watchers.push(subscription)
-                        }
+                        match subscription {
+                            WatchKind::Task(subscription) => {
+                                self.add_task_subscription(subscription);
+                            },
+                            WatchKind::TaskDetail(watch_request) => {
+                                self.add_task_detail_subscription(watch_request);
+                            },
+                        };
                     } else {
                         tracing::debug!("rpc channel closed, terminating");
                         return;
```

Comment:
style nit, take it or leave it: we could combine the `match` and `if let` here, to something like:
```rust
match subscription {
    Some(WatchKind::Task(subscription)) = {
         self.add_task_subscription(subscription);
    },
    Some(WatchKind::TaskDetail(watch_request)) => {
        self.add_task_detail_subscription(watch_request);
    },
    None => {
        tracing::debug!("rpc channel closed, terminating");
        return;
    },
}
```

not a big deal either way, but this might be a bit neater.